src/bp/core/services/telemetry/configs.ts

```diff
@@ -0,0 +1,170 @@
+import { buildSchema, TelemetryEvent } from 'common/telemetry'
+import { BotConfig } from 'core/config/bot.config'
+import { BotpressConfig } from 'core/config/botpress.config'
+import { ConfigProvider } from 'core/config/config-loader'
+import Database from 'core/database'
+import { calculateHash } from 'core/misc/utils'
+import { ModuleLoader } from 'core/module-loader'
+import { TelemetryRepository } from 'core/repositories/telemetry'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import defaultJsonBuilder from 'json-schema-defaults'
+import _ from 'lodash'
+import ms from 'ms'
+
+import { GhostService } from '..'
+import { BotService } from '../bot-service'
+import { JobService } from '../job-service'
+import { Config } from '../module/config-reader'
+
+import { TelemetryStats } from './telemetry-stats'
+
+const DEFAULT = '***default***'
+const REDACTED = '***redacted***'
+const SECRET_KEYS = [
+  'secret',
+  'pw',
+  'password',
+  'token',
+  'key',
+  'promoted_by',
+  'description',
+  'admins',
+  'email',
+  'connection'
+]
+
+interface BotConfigEvent {
+  botId: string
+  botConfigs: BotConfig
+}
+
+interface ModuleConfigEvent {
+  botId: string
+  module: string
+  configs: Config
+}
+
+@injectable()
+export class ConfigsStats extends TelemetryStats {
+  protected interval: number
+  protected url: string
+  protected lock: string
+
+  constructor(
+    @inject(TYPES.GhostService) ghostService: GhostService,
+    @inject(TYPES.Database) database: Database,
+    @inject(TYPES.LicensingService) licenseService: LicensingService,
+    @inject(TYPES.JobService) jobService: JobService,
+    @inject(TYPES.TelemetryRepository) telemetryRepo: TelemetryRepository,
+    @inject(TYPES.BotService) private botService: BotService,
+    @inject(TYPES.ModuleLoader) private moduleLoader: ModuleLoader,
+    @inject(TYPES.ConfigProvider) private config: ConfigProvider
+  ) {
+    super(ghostService, database, licenseService, jobService, telemetryRepo)
+    this.url = process.TELEMETRY_URL
+    this.lock = 'botpress:telemetry-configs'
+    this.interval = ms('7d')
+  }
+
+  protected async getStats(): Promise<TelemetryEvent> {
+    return {
+      ...buildSchema(await this.getServerStats(), 'server'),
+      event_type: 'configs',
+      event_data: {
+        schema: '1.0.0',
+        botConfigs: await this.getBotsConfigs(),
+        modulesConfigs: await this.getModulesConfigs(),
+        botpressConfig: await this.getBotpressConfigs()
+      }
+    }
+  }
+
+  private obfuscateSecrets(config, defaultConfig): any {
+    return _.reduce(
+      config,
+      (res, value, key) => {
+        if (SECRET_KEYS.find(x => key.toLowerCase().includes(x))) {
+          res[key] = _.isEqual(config[key], defaultConfig[key]) ? DEFAULT : REDACTED
+        } else if (!_.isArray(value) && _.isObject(value)) {
+          res[key] = this.obfuscateSecrets(value, defaultConfig[key])
```

Comment:
Missing a null check for the default config (in case the user have more configuration keys or old config)
![image](https://user-images.githubusercontent.com/42552874/89305651-ebc29080-d63c-11ea-84ab-fc13ec4c86c2.png)

`defaultConfig?.[key]`

Reply:
fixed: 96a61ffd9f449ee7caede738160504b83874146c

Reply:
changed for ES6 shorthand in c4e3bfef45be61c1bd7f7c8a05012c05addad9c2